core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -628,12 +565,12 @@ public void onConnectionClosedByServer(Status status) {
 
                 @Override
                 public void onInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, true);
+                  isIdle = false;
                 }
 
                 @Override
                 public void onNotInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, false);
+                  isIdle = true;
```

Comment:
I don't know how `ConcurrentHashMap` helps, since size() isn't atomic. I have trouble believing this part would be very hot, so I wouldn't necessarily push to get rid of the lock. It may be possible to use an atomic long and notice when it hits zero, but I expect even that may be a bit too interesting for little gain.